packages/gatsby/src/datastore/in-memory/indexing.ts

```diff
@@ -98,6 +98,18 @@ export const getGatsbyNodePartial = (
         fullNodeObject = getNode(node.id)!
       }
 
+      if (
+        dottedField.startsWith(`__gatsby_resolved.`) &&
+        !fullNodeObject.__gatsby_resolved
+      ) {
+        const typeName = fullNodeObject.internal.type
+        const resolvedNodes = store.getState().resolvedNodesCache.get(typeName)
+        const resolved = resolvedNodes?.get(fullNodeObject.id)
+        if (resolved !== undefined) {
+          fullNodeObject.__gatsby_resolved = resolved
+        }
+      }
+
```

Comment:
Main part of the fix - node might just not have `__gatsby_resolved` on it if it was GCed since materialization did run. This just check wether it has it and pull already resolved values from resolved node cache ( materialization saves results there and this should be main source of truth )